patcher.sh

```diff
@@ -0,0 +1,302 @@
+		fi
+
+		# Format music for Unix, aka lowercase it
+		# Finds every file, and if its not already lowercase, lowercases it.
+		find "$RESOURCES" -type f -exec \
+			bash -c 'target="{}";
+				cd "$(dirname "${target}")";
+				target="$(basename "${target}")";
+				! [[ "${target}" = "${target,,}" ]] && mv "${target}" "${target,,}"' \;
+
+		# Remove old lang folder
+		rm -R "$OUTPUT"/lang
+
+		chmod +x "$OUTPUT/runner"
+
+		if [ "$SYSTEMWIDE" = true ]; then
+			mkdir "$PREFIX/bin"
+			ln -s "$OUTPUT/runner" "$PREFIX/bin/am2r"
+		fi
+
+		# Create .desktop file
+		echo "Creating desktop file..."
+		cp "$SCRIPT_DIR/data/files_to_copy/icon.png" "$OUTPUT/icon.png"
+		desktopPath="$OUTPUT/AM2R.desktop"
+		if [ "$SYSTEMWIDE" = true ]; then
+			mkdir -p "$PREFIX/share/applications"
+			desktopPath="$PREFIX/share/applications/AM2R.desktop"
+		fi
+		cp "$SCRIPT_DIR/DesktopTemplate" "$desktopPath"
+
+		# replace [REPLACE] with OUTPUT.Replace("/", "\/") in desktop file
+		sed -i "s/\[REPLACE\]/${OUTPUT//\//\\\/}/" "$desktopPath"
+
+		# If we don't want to create an AppImage, we're done.
+		if [ "$APPIMAGE" = false ]; then
+			# For non-appimage, the desktop file should point to runner
+			sed -i "s/AM2R.AppImage/runner/" "$desktopPath"
+			echo ""
+			echo "The operation was completed successfully. See you next mission!"
+			return 0
+		fi
+
+		# Create AppImage
+		echo "Creating AppImage..."
+		cp -R --preserve=links "$SCRIPT_DIR/data/AM2R.AppDir" "$SCRIPT_DIR/AM2R.AppDir"
+		mv "$OUTPUT" "$SCRIPT_DIR/AM2R.AppDir/bin"
+		ARCH=x86_64 "$SCRIPT_DIR/utilities/appimagetool-x86_64.AppImage" -n AM2R.AppDir 2> /dev/null
+
+		mkdir "$OUTPUT"
+		mv "$SCRIPT_DIR/AM2R-x86_64.AppImage" "$OUTPUT/AM2R.AppImage"
+		mv "$SCRIPT_DIR/AM2R.AppDir/bin/icon.png" "$OUTPUT/icon.png"
+		if [ "$SYSTEMWIDE" = false ]; then
+			mv "$SCRIPT_DIR/AM2R.AppDir/bin/AM2R.desktop" "$desktopPath"
+		fi
+		rm -R "$SCRIPT_DIR/AM2R.AppDir"
+
+	elif [ "$OSCHOICE" = "android" ]; then
+		echo "Creating an APK for Android..."
+		apktoolPath="$SCRIPT_DIR/utilities/android/apktool.jar"
+		uberPath="$SCRIPT_DIR/utilities/android/uber-apk-signer.jar"
+
+		echo "Applying Android patch..."
+		xdelta3 -dfs "$OUTPUT/data.win" "$SCRIPT_DIR/data/droid.xdelta" "$OUTPUT/game.droid"
+
+		cp "$SCRIPT_DIR/data/android/AM2RWrapper.apk" "$SCRIPT_DIR/utilities/android/"
+		rm "$OUTPUT/D3DX9_43.dll" "$OUTPUT/AM2R.exe" "$OUTPUT/data.win"
+		mv "$OUTPUT" "$SCRIPT_DIR/utilities/android/assets"
+
+		if [ -f "$SCRIPT_DIR/data/android/AM2R.ini" ]; then
+			cp -p "$SCRIPT_DIR/data/android/AM2R.ini" "$SCRIPT_DIR/utilities/android/assets/"
+		fi
+
+		echo "Installing new datafiles..."
+		cp -R "$SCRIPT_DIR"/data/files_to_copy/* "$SCRIPT_DIR/utilities/android/assets"
+
+		if [ "$HQMUSIC" = true ]; then
+			cp "$SCRIPT_DIR"/data/HDR_HQ_in-game_music/*.ogg "$SCRIPT_DIR/utilities/android/assets/"
+		fi
+
+		echo "Packaging APK..."
+		echo "If Java JDK 8 or newer is not present, this will fail!"
```

Comment:
It'd be nice if this were to be checked instead.

Reply:
Same with xdelta too then? Since those patching processes fail currently as well if it is not installed.

Reply:
Done locally.